pkg/ddc/base/runtime.go

```diff
@@ -397,7 +406,7 @@ func GetRuntimeInfo(client client.Client, name, namespace string) (runtimeInfo R
 		if err != nil {
 			return runtimeInfo, err
 		}
-		runtimeInfo, err = BuildRuntimeInfo(name, namespace, common.EFCRuntime, datav1alpha1.TieredStore{}, WithMetadataList(efcRuntime.Spec.MetadataList))
+		runtimeInfo, err = BuildRuntimeInfo(name, namespace, common.EFCRuntime, datav1alpha1.TieredStore{}, WithMetadataList(GetMetadataListFromAnnotation(efcRuntime)))
```

Comment:
Please also modify the following funcs to make metadata list work when creating runtimes:

https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/alluxio/runtime_info.go#L33
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/efc/runtime_info.go
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/goosefs/runtime_info.go
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/jindofsx/runtime_info.go#L32
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/juicefs/runtime_info.go#L33
https://github.com/fluid-cloudnative/fluid/blob/master/pkg/ddc/thin/runtime_info.go#L33